microcosm_secretsmanager/loaders/base.py

```diff
@@ -0,0 +1,59 @@
+from json import loads
+from json.decoder import JSONDecodeError
+from warnings import warn
+
+from boto3 import Session
+from botocore.exceptions import ClientError
+
+
+class SecretsManagerLoader:
+    def __init__(self,
+                 environment=None,
+                 profile_name=None,
+                 region=None):
+
+        self.environment = environment
+        self.profile_name = profile_name
+        self.region = region
+
+    def __call__(self, metadata, version=None):
+        service = metadata.name
+        return self.get_secret_value(service, version)
+
+    def get_secret_value(self, service, version=None):
+        keyname = self.keyname(service)
+        client = self.make_client(service)
```

Comment:
Do we make a client per get? That seems wasteful.

Reply:
I'd agree except that the usual use case here is to call this once, on startup, and never again.